plugins/statistics.koplugin/calendarview.lua

```diff
@@ -364,6 +365,523 @@ function CalendarWeek:update()
+                self.check_container,
+                HorizontalSpan:new{ w = Size.padding.default },
+                OverlapGroup:new{
+                    dimen = Geom:new{ w = title_max_width, h = self.height},
+                    allow_mirroring = false,
+                    FrameContainer:new{
+                        width = title_max_width,
+                        height = self.height - 2 * Size.padding.small,
+                        padding = 0,
+                        padding_left = self.padding,
+                        padding_right = self.padding,
+                        background = bgcolor,
+                        bordersize = Size.border.thin,
+                        overlap_offset = { 0, Size.padding.small },
+                        LeftContainer:new{
+                            dimen = Geom:new{
+                                w = title_max_width,
+                                h = self.height - 2 * Size.padding.small,
+                            },
+                            TextWidget:new{
+                                text = self.item[1],
+                                max_width = title_max_width - 2 * self.padding,
+                                face = self.face,
+                                bgcolor = bgcolor,
+                                fgcolor = fgcolor,
+                                padding = 0,
+                                bordersize = Size.border.thin,
+                            }
+                        }
+                    }
+                },
+                FrameContainer:new{
+                    width = self.value_width,
+                    padding = 0,
+                    padding_left = Size.padding.default,
+                    bordersize = 0,
+                    LeftContainer:new{
+                        dimen = Geom:new{ w = self.value_width, h = self.height},
+                        padding = 0,
+                        TextWidget:new{
+                            text = self.item[2],
+                            max_width = self.value_width,
+                            face = self.face
+                        }
+                    }
+                }
+            }
+        }
+    }
+    checked_widget:free()
+    self[1].invert = self.invert
+end
+
+function BookDailyItem:onTap(_, ges)
+    local x_intersect = function()
+        local dimen = self.checkmark_widget.dimen
+        return ges.pos.x >= dimen.x - dimen.w and ges.pos.x <= dimen.x + 2 * dimen.w
+    end
+    if self.item.check_cb and x_intersect() then
+        self.item:check_cb()
+        self.checkmark_widget = CheckMark:new{
+            checked = self.item.checked,
+        }
+        self.check_container[1] = self.checkmark_widget
+        UIManager:setDirty(self, function()
+            return "ui", self.check_container.dimen
+        end)
+    elseif self.item.callback then
+        self.item:callback()
+    end
+    return true
+end
+
+local CalendarDayDetail = FocusManager:extend{
```

Comment:
May be name this `CalenderDayView` ? to make it clearer it's a fullscreen widget like CalendarView and not a sub-component widget?